src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1161,13 +1161,18 @@ internal bool FilterAndAppendAndFreeDiagnostics(DiagnosticBag accumulator, ref D
         /// Filter out warnings based on the compiler options (/nowarn, /warn and /warnaserror) and the pragma warning directives.
         /// </summary>
         /// <returns>True when there is no error.</returns>
-        internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming)
+        internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming, ISet<int> exclude)
```

Comment:
>ISet<int> [](start = 110, length = 9)

Is there a real advantage in using an interface for this parameter? Interface dispatch is usually more expensive.